loopout.go

```diff
@@ -638,50 +664,114 @@ func (s *loopOutSwap) payInvoiceAsync(ctx context.Context,
 	paymentStateCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	payStatusChan, payErrChan, err := s.lnd.Router.SendPayment(
-		paymentStateCtx, req,
+	// Attempt to acquire and initialize the routing plugin.
+	routingPlugin, err := AcquireRoutingPlugin(
+		ctx, pluginType, *s.lnd, target, amt,
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	for {
-		select {
-		// Payment advanced to the next state.
-		case payState := <-payStatusChan:
-			s.log.Infof("Payment %v: %v", hash, payState)
-
-			switch payState.State {
-			case lnrpc.Payment_SUCCEEDED:
-				return &payState, nil
-
-			case lnrpc.Payment_FAILED:
-				return &payState, nil
+	var (
```

Comment:
Could we pull out the following logic into a function? Also makes testing easier.

Reply:
Cool. Yeah usually I'd prefer small functions, sort like sticking to the SOLID principle. This type of decision is hard though, as it quickly becomes vague about what's the industry's best practice vs what's a personal taste😂 My take is that it's better to orginize code to reduce cognitive load. For instance the `StatusLoop` below has a simple task, to track the payment state and report the result, seems can be abstracted into a function like `processPayStatus`? Also I see we have two important logical branches here, that is whether `routingPlugin` is nil or not, which determines whether we want to have a `maxRetries` of 1 or more. It probably could be better if we could organize the branches like,

```
if routingPlugin == nil {
	sendPaymentOnce()
} else {
	sendPaymentMultipleTimes()
}
```
But in the end, what do I know. Since I'm not the author and do not know the details as comprehensively, it's up to you to make the decisions:)